### ISSUE TITLE
Reorder directories to better conform to Python package standards

### ISSUE THREAD

**Issue by yantosca** [geoschem/gcpy#249]
Right now we have the following directory structure:

```console
gcpy
    |
    gcpy
    |
    benchmark
    |       |
    |      modules
    |
    examples
            |
            several  other subdirectories not listed
```
But according to Python packaging best practice we should use a directory structure such as:
```console
gcpy
    |
    gcpy
       |
       benchmark
       |       |
       |       modules
       |
       examples
              |
              several other subdirectories not listed
```
Key features:
1. The root package folder `gcpy` now has a `src` folder, where all source code goes
2. `gcpy/gcpy` is now moved to `gcpy/src/gcpy`
3. All other folders (benchmark, examples) are now subfolders of `gcpy/src/gcpy`.

Background: comment by @lizziel in issue, see https://github.com/geoschem/gcpy/issues/245#issuecomment-1664514951

**Comment by yantosca** [comment]
Also note, this will involve renaming the current `gcpy/benchmark.py` to something like `benchmark_funcs.py` in order to avoid a namespace collision with the `benchmark` folder.

**Comment by yantosca** [comment]
We can now close this issue as PR #250 has been merged into the GCPy 1.4.0 development stream.
